p2p_wallet/Scenes/Main/Buy/BuyCoordinator.swift

```diff
@@ -143,12 +144,41 @@ final class BuyCoordinator: Coordinator<Void> {
             }
         }).store(in: &subscriptions)
 
-        viewModel.coordinatorIO.license.sink(receiveValue: { [weak self] url in
-            let vc = SFSafariViewController(url: url)
-            vc.modalPresentationStyle = .automatic
-            viewController.present(vc, animated: true)
-        }).store(in: &subscriptions)
-
+        viewModel.coordinatorIO.license
+            .sink(receiveValue: { url in
+                let vc = SFSafariViewController(url: url)
+                vc.modalPresentationStyle = .automatic
+                viewController.present(vc, animated: true)
+            })
+            .store(in: &subscriptions)
+        viewModel.coordinatorIO.close
+            .sink(receiveValue: { [unowned self] in
+                navigationController.popViewController(animated: true)
+            })
+            .store(in: &subscriptions)
+        viewModel.coordinatorIO.chooseCountry
+            .sink(receiveValue: { [weak self] selectedCountry in
+                guard let self else { return }
```

Comment:
Can we move inside logic to a separate coordinator and make

viewModel.coordinatorIO.chooseCountry.flatMap { selectedCountry in self.coordinate(to: SelectCountryCoordinator() }.sink { selectCountryResult in ... }

Reply:
Yes. Thanks. It's already been discussed.

Reply:
refactoring https://p2pvalidator.atlassian.net/browse/PWN-8175